app/src/test/java/com/onewelcome/showcaseapp/viewmodel/ImplicitAuthenticationViewModelTest.kt

```diff
@@ -0,0 +1,210 @@
+import com.onewelcome.core.util.TestConstants
+import com.onewelcome.core.util.TestConstants.TEST_USER_PROFILES
+import com.onewelcome.core.util.TestConstants.TEST_USER_PROFILE_1
+import com.onewelcome.showcaseapp.fakes.OmiSdkEngineFake
+import com.onewelcome.showcaseapp.feature.sdkreset.SdkResetViewModel
+import com.onewelcome.showcaseapp.feature.userauthentication.implicitauthentication.ImplicitAuthenticationViewModel
+import com.onewelcome.showcaseapp.feature.userauthentication.implicitauthentication.ImplicitAuthenticationViewModel.UiEvent.StartImplicitAuthentication
+import com.onewelcome.showcaseapp.feature.userauthentication.implicitauthentication.ImplicitAuthenticationViewModel.UiEvent.UpdateSelectedUserProfile
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.HiltTestApplication
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+
+@HiltAndroidTest
+@Config(application = HiltTestApplication::class)
+@RunWith(RobolectricTestRunner::class)
+class ImplicitAuthenticationViewModelTest {
+  @get:Rule
+  val hiltRule = HiltAndroidRule(this)
+
+  @Inject
+  lateinit var omiSdkEngineFake: OmiSdkEngineFake
+
+  @Inject
+  lateinit var oneginiClientMock: OneginiClient
+
+  @Inject
+  lateinit var isSdkInitializedUseCase: IsSdkInitializedUseCase
+
+  @Inject
+  lateinit var implicitAuthenticationUseCase: ImplicitAuthenticationUseCase
+
+  @Inject
+  lateinit var getUserProfilesUseCase: GetUserProfilesUseCase
+
+  @Inject
+  lateinit var getImplicitlyAuthenticatedUserProfileUseCase: GetImplicitlyAuthenticatedUserProfileUseCase
+  private val userClientMock: UserClient = mock()
+  private val mockOneginiImplicitTokenRequestError: OneginiImplicitTokenRequestError = mock()
+  lateinit var viewModel: ImplicitAuthenticationViewModel
+
+  @Before
+  fun setup() {
+    hiltRule.inject()
+    viewModel = ImplicitAuthenticationViewModel(
+      isSdkInitializedUseCase,
+      getUserProfilesUseCase,
+      implicitAuthenticationUseCase,
+      getImplicitlyAuthenticatedUserProfileUseCase
+    )
+  }
+
+  @Test
+  fun `Given sdk is not initialized, When LoadData event is sent, Then default state should be returned`() {
+    val expectedState = viewModel.uiState.copy(null, false, emptyList(), emptySet(), null, false)
+
+    assertThat(viewModel.uiState).isEqualTo(expectedState)
+  }
+
+  @Test
+  fun `Given sdk is initialized, When LoadData event is sent, Then data should be updated`() {
+    val expectedState = viewModel.uiState.copy(isSdkInitialized = true)
+    mockSdkInitialized()
+    initializeViewModel()
```

Comment:
It's a good practice to separate visually the Given - When - Then steps by empty spaces. So here that would be 
```
   val expectedState = viewModel.uiState.copy(
      isSdkInitialized = true,
      userProfiles = TEST_USER_PROFILES,
      selectedUserProfile = TEST_USER_PROFILES.first(),
      isAuthenticateButtonEnabled = true,
      implicitlyAuthenticatedUserProfile = TEST_USER_PROFILES.first()
    )
    mockSdkInitialized()
    mockUserClient()
    mockUserProfiles()
    mockImplicitlyAuthenticatedProfile()
    
    initializeViewModel()

    assertThat(viewModel.uiState).isEqualTo(expectedState)

Please align the rest of test cases.